study1/week6/NumDecodings.java

```diff
@@ -0,0 +1,33 @@
+class NumDecodings {
+    public int numDecodings(String s) {
+
+        int n = s.length();
+        int[] dp = new int[n + 1];
```

Comment:
You can make this constant space by only keeping track of the last two elements.